drivers/sdl/joypad_sdl.cpp

```diff
@@ -164,9 +165,18 @@ void JoypadSDL::process_events() {
 
 				sdl_instance_id_to_joypad_id.insert(sdl_event.jdevice.which, joy_id);
 
-				// Skip Godot's mapping system because SDL already handles the joypad's mapping
 				Dictionary joypad_info;
-				joypad_info["mapping_handled"] = true;
+				joypad_info["mapping_handled"] = true; // Skip Godot's mapping system because SDL already handles the joypad's mapping
+				if (SDL_GetJoystickPlayerIndex(joy) >= 0) {
+					// For XInput controllers SDL_GetJoystickPlayerIndex returns the XInput user index.
+					joypad_info["xinput_index"] = itos(SDL_GetJoystickPlayerIndex(joy));
+				}
```

Comment:
Should we name the key to be `player_index` instead, as it is named by SDL? I wonder if we could do like `SDL` and just return `0` on platforms that it doesn't work on.

```suggestion
				joypad_info["player_index"] = itos(SDL_GetJoystickPlayerIndex(joy));
```

Reply:
Just saw that it was named like this before. Could we add both for the time, marking "xinput_index" as deprecated?

 cc. @clayjohn @mhilbrunner

Reply:
What is the benefit of renaming it? Just to match the naming in SDL?

Reply:
Maybe it's not that urgent. I just felt that if SDL has a different name, then it would maybe make more sense to follow its standard.